### PR TITLE
Adds of_string function to Csexp.

### PR DESCRIPTION
**Pull request by staronj**
Signed-off-by: Jakub Staron <jstaron@janestreet.com>

### CODE REVIEW

File: src/stdune/csexp.mli [hunk 1]
```diff
@@ -1,12 +1,15 @@
 type t = Sexp.t
 
+(** [to_string sexp] converts S-expression [sexp] to a string in canonical
+  form. *)
+val to_string : t -> string
+
+(** [of_string string] parses [string] into S-expression *)
+val of_string : string -> (t, string) Result.t
```
Comment (rgrinberg) [inline src/stdune/csexp.mli]: If it's doing parsing, then it should just be called `parse` or `parse_string`. I recall we had this discussion regarding `Path` before (see Path_intf)

File: src/stdune/csexp.ml [hunk 2]
```diff
@@ -74,3 +74,5 @@ let to_string sexp =
   let buf = buffer () in
   to_buffer sexp ~buf;
   Buffer.contents buf
+
+let of_string string = parse (Stream.of_string string)
```
Comment (aalekseyev) [inline src/stdune/csexp.ml]: I think this function should fail if there's a non-empty suffix remaining after parsing. As it stands, I think it ignores all input after the first valid csexp.

**Comment by aalekseyev** [comment]
Not sure why github swallowed my comments, but I can type them again.

1. You could use bind (`>>=`) in `parse_string`
2. I don't think the demonstration that `parse_string` agrees with `parse` on all valid inputs is necessary: let's just test `parse_string` in simple cases and only call `parse` when we have extra input.

**Comment by rgrinberg** [comment]
> You could use bind (>>=) in parse_string

To add to this, `let*` with the appropriate module opened is preferable.

**Pull request merged**